azkaban-web-server/src/main/java/azkaban/scheduler/CustomDeserializerDelegate.java

```diff
@@ -0,0 +1,627 @@
+package azkaban.scheduler;
```

Comment:
Ugh... at the first blush, this is way, *way* too much Quartz internal logic dumped into Azkaban. What happens when Quartz changes `StdJDBCDelegate`? Will we have to forever maintain this piece of code in sync with Quartz changes? Personally I would rather purge the DB and tell users to recreate their schedules if necessary than add this to Azkaban and try to explain to anyone why we did this -- even long after no one has any `...228L` UIDs left in their databases anymore. 
Are we sure there is no better way?

Reply:
I must admit though, the more I look into this, the more I respect what you've accomplished here... a true hack.

For my education: why can't we directly extend `StdJDBCDelegate`, overwrite `getObjectFromBlob` *only* with your filtering version, tell Quartz (via `org.quartz.jobStore.driverDelegateClass` configuration property) to use that delegate from get-go ...and call it good? 

Extra benefit is that anyone who hasn't been exposed to the new UID, or who purged their DB of it, either deliberately or over time, doesn't need to bother and can just use straight `StdJDBCDelegate` again.

Reply:
There are solutions in the vein of modifying the Quartz database, but that would require manual intervention and worse it would require every user affected by this issue to get into the guts of Quartz. I'm not even sure what such a solution would look like. Clearing the Quartz database could easily be a problematic amount of data loss for any organization using this at any scale in production. Either way, those remain as backup options in case this feature doesn't work. This solution attempts to make it painless for the user. 

Yes, there are many things that could go wrong. Using reflection to change private fields and then doing a search-and-replace on a binary object serialization stream is obviously not ideal. I do believe this will work for pretty much everyone using RDBS-backed Quartz, which should be most users. Quartz code itself is fairly stable here, so I don't expect it to change anytime soon.

Reply:
I wrapped QuartzScheduler::enableSerializationHack with an extra try/catch to defend against anything unexpected side effects. I also added a comment at the header of this class (renamed to CustomQuartzDeserializer) to discuss the scope and intent.

Reply:
Sorry, again I haven't been clear. I am not advocating changing the DB. I am questioning the massive reflection-based wrapper for StdJDBCDelegate that has to implement every method. Can we essentially add an implementation of StdJDBCDelegate as if it was another `MSSQLDelegate` or `OracleDelegate`, and only override `getObjectFromBlob` in this custom class? Then inject that version via the `org.quartz.jobStore.driverDelegateClass` Azkaban property? Is there something that I am missing that makes the wrapper approach necessary and the subclass approach impossible?

Reply:
For some reason your second comment in this thread only just now appeared for me. Such an approach might work, I'll look into it. It would avoid reflection. However, it would need to be manually configured. Also, we'd need to add and read another property to figure out what original driverDelegateClass we are supposed to wrap (or make a lot of subclasses to support each RDBS type). I'm a little worried about things going wrong with that extra logic, but it's certainly no worse than reflection-based injection.

Reply:
I've made changes to allow the explicit configuration of CustomQuartzDeserializer and separated out the reflection-based injector into QuartzHacker. This allows the zero-configuration bug mitigation to exist and be managed separately from the core fix. If triggered, it will output at warn level and explanation that guides the user to explicitly configure the fix.